core/src/main/scala/akka/kafka/internal/ConfigSettings.scala

```diff
@@ -4,32 +4,31 @@
  */
 package akka.kafka.internal
 
-import com.typesafe.config.{Config, ConfigObject, ConfigValue}
+import java.util
+
+import com.typesafe.config.{Config, ConfigObject}
+
+import scala.collection.JavaConverters.asScalaSetConverter
 
 /**
  * INTERNAL API
  */
 private[kafka] object ConfigSettings {
 
   def parseKafkaClientsProperties(config: Config): Map[String, String] = {
-    def collectKeys(c: ConfigObject, prefix: String, keys: Set[String]): Set[String] = {
```

Comment:
I think it's good to improve it, thanks.

To clarify, for the archives, the old grows stack per prop level, not per property. E.g. aaa.bbb.ccc.ddd.eee.fff is 6 levels.